administrator/components/com_media/controllers/file.json.php

```diff
@@ -172,7 +171,7 @@ public function upload()
 		}
 
 		// Trigger the onContentAfterSave event.
-		$dispatcher->trigger('onContentAfterSave', array('com_media.file', &$object_file, true));
+		$dispatcher->dispatch('onContentAfterSave', array('com_media.file', &$object_file, true));
```

Comment:
I think this needs to go through `JApplication` like the call above - `$dispatcher` isn't defined anymore in this file